router/batchrouter/handle_lifecycle.go

```diff
@@ -53,7 +53,10 @@ func (brt *Handle) Setup(
 ) {
 	brt.destination = destination
 	brt.destType = destination.DestinationDefinition.Name
-	asyncdestinationmanager, err := asyncdestinationmanager.NewManager(destination, backendConfig)
+	opts := common.AsyncDestinationOpts{
+		MaxUploadSize: brt.maxUploadLimit,
+	}
+	asyncdestinationmanager, err := asyncdestinationmanager.NewManager(destination, backendConfig, opts)
 	if err != nil {
 		panic(fmt.Errorf("unable to initialize async destination manager for: %w", err))
```

Comment:
avoid panic

Reply:
not sure how to deal with this, would need guidance. @koladilip

Reply:
My two cents, we can `NOOP` instead of `panic`. Similar to what we did for error-reporting.

A thumb rule for `panic` would be if the system doesn't have meaning without existence of an entity, only then we `panic`.

For example
Let's assume we have a feature. The feature makes use of a table(whose migration is done programatically at start of the application) for all of it's operations, even before the starting the feature functionality we need the table to be present. In this case, we may do a `panic`

Reply:
We can retry after sometime to create new manager using exponential  backoff method, if all retries failed then log error and set up a metric so that we can create a  P0 alert for it. @cisse21 is there any better way to avoid this panic?